client/lib/signup/step-actions/index.js

```diff
@@ -486,6 +496,102 @@ export function setIntentOnSite( callback, { siteSlug, intent } ) {
 			callback( [ errors ] );
 		} );
 }
+export function addWithThemePlanToCart( callback, dependencies, stepProvidedItems, reduxStore ) {
+	const { cartItem } = stepProvidedItems;
+
+	/**
+	 * If the user selected the free option, then we should abort the checkout part.
+	 */
+	if ( ! cartItem ) {
+		defer( callback );
+		return;
+	}
+
+	reduxStore.dispatch( requestTheme( dependencies.theme, 'wpcom', getLocaleSlug() ) ).then( () => {
+		const state = reduxStore.getState();
+		const themeSlug = dependencies.theme;
+		const isExternallyManagedTheme = getIsExternallyManagedTheme( state, themeSlug );
+
+		if ( isExternallyManagedTheme ) {
+			addExternalManagedThemeToCart(
+				state,
+				reduxStore.dispatch,
+				themeSlug,
+				dependencies.siteSlug,
+				cartItem
+			).then( () => {} );
+		} else {
+			addPlanToCart( callback, dependencies, stepProvidedItems, reduxStore );
+		}
+	} );
+}
+
+const isLoadingCart = ( isLoading ) => ( dispatch ) => {
```

Comment:
Nitpick (non-blocking): The name of this action resembles a selector. Maybe `setIsLoadingCard` is clearer?